### PR TITLE
[SIMULATION] [Clang]Cleanup clang-analyzer warnings

### PR DESCRIPTION
**Pull request by smuzaffar**
This fixes clang static analyzer warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-10-02-1100/el8_amd64_gcc12/build-logs/

### CODE REVIEW

File: SimG4Core/CustomPhysics/src/CMSSQNeutronAnnih.cc [hunk 1]
```diff
@@ -87,7 +87,7 @@ G4double CMSSQNeutronAnnih::momDistr(G4double x_in) {
 
   //now interpolate the above points for x_in
   G4double result = 9999;
-  for (int i = 0; i < n_entries; i++) {
+  for (int i = 1; i < n_entries; i++) {
```
Comment (smuzaffar) [inline SimG4Core/CustomPhysics/src/CMSSQNeutronAnnih.cc:90]: clang-analyzer was complaining that for `i=0`, we are trying to access `CDF_k[-1] and x[-1`.  So I started the loop from `1`. @cms-sw/simulation-l2 please check if this is correct

File: SimG4Core/Geometry/src/CMSG4CheckOverlap.cc [hunk 2]
```diff
@@ -240,6 +240,8 @@ void CMSG4CheckOverlap::makeReportForOverlaps(std::ofstream& fout,
         if (!reg) {
           fout << "### NO G4Region found - EXIT"
                << "\n";
+          if (gdml)
+            delete gdml;
```
Comment (smuzaffar) [inline SimG4Core/Geometry/src/CMSG4CheckOverlap.cc:244]: this was a possible memory leak if we fail to get region

File: SimG4Core/CustomPhysics/src/CMSSQNeutronAnnih.cc (hunk 1 above)
Comment (civanch) [reply SimG4Core/CustomPhysics/src/CMSSQNeutronAnnih.cc:90]: I am not sure what is real range for x_in but I would use following code, which is valid in all cases:

  if (x_in <= 0.0) { return 0.0; }
  if (x_in >= 1.0) { return CDF_k[n_entries - 1];}
  for (int i = 1; i < n_entries; i++) {
    if (x[i] >= x_in) {
      return (CDF_k[i] - CDF_k[i - 1]) / (x[i] - x[i - 1]) * (x_in - x[i - 1]) + CDF_k[i - 1];
    }
  }
  return 0.0;

Comment (smuzaffar) [reply SimG4Core/CustomPhysics/src/CMSSQNeutronAnnih.cc:93]: done

File: SimG4Core/Geometry/src/CMSG4CheckOverlap.cc (hunk 2 above)
Comment (smuzaffar) [reply SimG4Core/Geometry/src/CMSG4CheckOverlap.cc]: done, I have removed the `if (!reg){}` block

**Comment by smuzaffar** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**